src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -155,20 +155,12 @@ internal void AppendMetadata(MethodBodyStreamEncoder methodBodyEncoder)
                     _metadataBuilder.AddTypeLayout(typeHandle, (ushort)typeBuilder.PackingSize, (uint)typeBuilder.Size);
                 }
 
-                if (typeBuilder._interfaces != null)
-                {
-                    foreach (Type iface in typeBuilder._interfaces)
-                    {
-                        _metadataBuilder.AddInterfaceImplementation(typeHandle, GetTypeHandle(iface));
-                        // TODO: need to add interface mapping between interface method and implemented method
-                    }
-                }
-
                 if (typeBuilder.DeclaringType != null)
                 {
                     _metadataBuilder.AddNestedType(typeHandle, (TypeDefinitionHandle)GetTypeHandle(typeBuilder.DeclaringType));
                 }
 
+                AddInterfaceImplementations(typeBuilder, typeHandle);
                 WriteCustomAttributes(typeBuilder._customAttributes, typeHandle);
```

Comment:
Is there a reason why some of these methods are called `Add*` and some are called `Write*`?

Reply:
no reason, will update